### PR TITLE
Added more mostly internal code comments.

### PR DESCRIPTION
**Pull request by ICGog**

### CODE REVIEW

File: src/node/lattice.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ impl RunnableEvent {
         }
     }
 
-    /// Add a timestamp to the event.
+    /// Adds a `Timestamp` to the event.
```
Comment (pschafhalter) [inline src/node/lattice.rs:40]:
```suggestion
    /// Adds a [`Timestamp`] to the event.
```

**Review by pschafhalter — COMMENTED** [review]
See the minor change. Otherwise LGTM

File: src/dataflow/connect.rs [hunk 2]
```diff
@@ -64,6 +66,8 @@ where
     write_stream
 }
 
+/// Adds a parallel sink operator, which receives data on input read streams and directly
+/// interacts with external systems.
```
Comment (sukritkalra) [inline src/dataflow/connect.rs:70]: Can we refer to the actual types of the operators in these documentations so developers can click and see the interface?

File: src/dataflow/stream/errors.rs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 use crate::communication::{CommunicationError, TryRecvError};
 
-/// Errors raised by reading from a `ReadStream`.
+/// Error raised when something went wrong while reading from a `ReadStream`.
```
Comment (sukritkalra) [inline src/dataflow/stream/errors.rs:3]: Errors?

File: python/setup.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 package_dir = os.path.dirname(__file__)
 
 # Put all package dependencies here
-requires = ["fire", "numpy"]
+requires = ["fire", "numpy==1.19.0"]
```
Comment (sukritkalra) [inline python/setup.py:9]: Is there any reason we force a numpy version?

**Review by sukritkalra — COMMENTED** [review]
I just remembered this by seeing your changes in `src/dataflow/time.rs`, but can you also please double check the `cmp` function in that file? 

Specifically, what is the correct ordering of two `Timestamp::Bottom`s or two `Timestamp::Top`s?

File: python/setup.py (hunk 4 above)
Comment (ICGog) [reply python/setup.py:9]: Yes, I'm forcing numpy to a version that still supports Python 3.6. Otherwise, our CI fails to build for Python 3.6.

Comment (ICGog) [reply python/setup.py:9]: Once we fix the Rust tests, the CI should be fine again.

**Review by sukritkalra — APPROVED** [review]
LGTM!

**Pull request merged**